python/kserve/kserve/model.py

```diff
@@ -74,20 +100,47 @@ def __init__(self, name: str):
         self.timeout = 600
         self._http_client_instance = None
         self._grpc_client_stub = None
+        self.log_latency = False
 
     async def __call__(self, body, model_type: ModelType = ModelType.PREDICTOR, headers: Dict[str, str] = None):
-        payload = await self.preprocess(body, headers) if inspect.iscoroutinefunction(self.preprocess) \
-            else self.preprocess(body, headers)
+        request_id = headers.get("X-Request-Id")
+
+        # latency vars
+        preprocess_ms = 0
+        explain_ms = 0
+        predict_ms = 0
+        postprocess_ms = 0
+
+        with PRE_HIST_TIME.time():
+            start = time.time()
+            payload = await self.preprocess(body, headers) if inspect.iscoroutinefunction(self.preprocess) \
+                else self.preprocess(body, headers)
+            preprocess_ms = get_latency_ms(start, time.time())
         payload = self.validate(payload)
         if model_type == ModelType.EXPLAINER:
-            response = (await self.explain(payload, headers)) if inspect.iscoroutinefunction(self.explain) \
-                else self.explain(payload, headers)
+            with EXPLAIN_HIST_TIME.time():
+                start = time.time()
+                response = (await self.explain(payload, headers)) if inspect.iscoroutinefunction(self.explain) \
+                    else self.explain(payload, headers)
+                explain_ms = get_latency_ms(start, time.time())
         elif model_type == ModelType.PREDICTOR:
-            response = (await self.predict(payload, headers)) if inspect.iscoroutinefunction(self.predict) \
-                else self.predict(payload, headers)
+            with PREDICT_HIST_TIME.time():
+                start = time.time()
+                response = (await self.predict(payload, headers)) if inspect.iscoroutinefunction(self.predict) \
+                    else self.predict(payload, headers)
+                predict_ms = get_latency_ms(start, time.time())
         else:
             raise NotImplementedError
-        response = self.postprocess(response)
+
+        with POST_HIST_TIME.time():
+            start = time.time()
+            response = self.postprocess(response)
+            postprocess_ms = get_latency_ms(start, time.time())
+
+        if validate_log_latency(self.log_latency):
```

Comment:
even performance is not a concern here it is also not ideal that if log_latency is not valid then it throws exception on every request.

Reply:
moved